src/reducers/templates.js

```diff
@@ -0,0 +1,15 @@
+import Immutable from 'immutable'
+import { actionReducer } from './utils'
+
+const templates = actionReducer(Immutable.fromJS({ templates: {}, loadInProgress: true }), {
+  ADD_TEMPLATES (state, { payload: { templates } }) {
+    const updates = {}
```

Comment:
Could you please wrap this peace of code into some sort of `groupById` utility function? I believe I saw this piece of code copy repeated in many places.